src/test/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/publicip/v5_4/OpenNebulaPublicIpPluginTest.java

```diff
@@ -139,296 +128,463 @@ public void testHasFailedUnsuccessful() {
-				.willReturn(client);
+		Mockito.when(this.virtualNetwork.delete()).thenReturn(this.response);
+		Mockito.when(this.response.isError()).thenReturn(false);
 
-		String securityGroupId = STRING_ID_ONE;
-		SecurityGroup securityGroup = Mockito.mock(SecurityGroup.class);
-		BDDMockito.given(OpenNebulaClientUtil.getSecurityGroup(Mockito.eq(client), Mockito.eq(securityGroupId)))
-				.willReturn(securityGroup);
+		// exercise
+		this.plugin.doDeleteInstance(client, this.publicIpOrder);
 
-		OneResponse response = Mockito.mock(OneResponse.class);
-		Mockito.when(securityGroup.delete()).thenReturn(response);
-		Mockito.when(response.isError()).thenReturn(true);
+        // verify
+		PowerMockito.verifyStatic(OpenNebulaClientUtil.class, Mockito.times(TestUtils.RUN_ONCE));
+		OpenNebulaClientUtil.getVirtualNetwork(Mockito.eq(this.client), Mockito.eq(this.instanceId));
+
+		Mockito.verify(this.virtualNetwork, Mockito.times(TestUtils.RUN_ONCE)).delete();
+		Mockito.verify(this.response, Mockito.times(TestUtils.RUN_ONCE)).isError();
+	}
+
+	// test case: when invoking doDeleteInstance with invalid parameters the plugin should
+	// throw an UnexpectedException
+	@Test
+	public void testDoDeleteInstanceFail() throws InstanceNotFoundException, InvalidParameterException,
+			UnauthorizedRequestException {
+		// set up
+		Mockito.when(this.virtualNetwork.delete()).thenReturn(this.response);
+		Mockito.when(this.response.isError()).thenReturn(true);
+		Mockito.when(this.response.getErrorMessage()).thenReturn(STRING_ID_ONE);
 
 		// exercise
-		this.plugin.deleteSecurityGroup(client, securityGroupId);
+        try {
+			this.plugin.doDeleteInstance(this.client, this.publicIpOrder);
+			Assert.fail();
+		} catch (UnexpectedException e) {
+        	Assert.assertEquals(STRING_ID_ONE, e.getMessage());
+		}
 
 		// verify
-		Mockito.verify(response, Mockito.times(1)).getErrorMessage();
+		PowerMockito.verifyStatic(OpenNebulaClientUtil.class, Mockito.times(TestUtils.RUN_ONCE));
+		OpenNebulaClientUtil.getVirtualNetwork(Mockito.eq(this.client), Mockito.eq(this.instanceId));
+
+		Mockito.verify(this.virtualNetwork, Mockito.times(TestUtils.RUN_ONCE)).delete();
+		Mockito.verify(this.response, Mockito.times(TestUtils.RUN_ONCE)).isError();
+		Mockito.verify(this.response, Mockito.times(TestUtils.RUN_ONCE)).getErrorMessage();
 	}
-	
-	// test case: When calling the detachNetworkInterfaceConnected method and cannot
-	// detach the NIC of the virtual machine passed by parameter, this will fail.
+
+	// test case: when invoking getInstance with valid public ip order and cloud user, the plugin
+	// should instantiate the respective fogbow public ip instance and return it
 	@Test
-	public void testDetachNetworkInterfaceConnectedUnsuccessfully()
-			throws InvalidParameterException, UnauthorizedRequestException, InstanceNotFoundException {
+	public void testGetInstance() throws FogbowException {
```

Comment:
This test has a runtime error...
Please check and correct.

Reply:
done